csharp/protos/google/protobuf/field_presence_test.proto

```diff
@@ -0,0 +1,24 @@
+syntax = "proto3";
+
+package Google.ProtocolBuffers.TestProtos.FieldPresence;
+
+option java_package = "com.google.protobuf";
+option java_outer_classname = "FieldPresenceTestProto";
+option java_generate_equals_and_hash = true;
+
+message TestAllTypes {
```

Comment:
I wouldn't call this TestAllTypes, given that it doesn't have all the types... why not TestFieldPresence?